packages/nimble-components/src/table-column/mixins/fractional-width-column.ts

```diff
@@ -0,0 +1,56 @@
+import { attr, nullableNumberConverter } from '@microsoft/fast-element';
+import type { TableColumn } from '../base';
+import { defaultMinPixelWidth, defaultFractionalWidth } from '../base/types';
+
+// Pick just the relevant properties the mixin depends on (typescript complains if the mixin declares private / protected base exports)
+type SizedTableColumn = Pick<
+TableColumn,
+'internalFractionalWidth' | 'internalMinPixelWidth'
+>;
+// prettier-ignore
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type SizedTableColumnConstructor = abstract new (...args: any[]) => SizedTableColumn;
+
+// As the returned class is internal to the function, we can't write a signature that uses is directly, so rely on inference
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
+export function mixinFractionalWidthColumnAPI<
```

Comment:
Could we document this mixin pattern in our CONTRIBUTING doc?

Reply:
Eventually, yes. I would prefer to let this PR in before spending time figuring out how to document this properly.

Reply:
Plan discussed in person:
- Meyer to create a follow up PR adding a very brief description of the general pattern to CONTRIBUTING
- Jesse captured a requirement to document specific mixins as part of [documentation for adding a new column type](https://github.com/ni/nimble/issues/976#issuecomment-1464505992)